packages/lib-core/src/store/PluginLoader.ts

```diff
@@ -304,10 +343,146 @@ export class PluginLoader {
+  ) {
+    const pluginName = pluginLoadData.manifest.name;
+
+    this.plugins.set(pluginName, pluginLoadData);
+    const initPromise = new Promise<void>((resolve, reject) => {
+      const script = getDocument().createElement('script');
+
+      script.src = resolveURL(baseURL, pluginLoadData.manifest.entryScript);
+      script.async = true;
+
+      script.onload = async (event) => {
+        if (onLoad) {
+          onLoad(event);
+        }
+        if (pluginLoadData.manifest.libraryType === 'var') {
+          await this.initializeVarModule(pluginName);
+        }
+        return resolve();
+      };
+
+      script.onerror = (event) => {
+        if (onError) {
+          onError(event);
+        }
+        return reject(new Error('Unable to initialize remote module'));
+      };
+
+      getDocument().head.appendChild(script);
+
+      consoleLogger.info(`Loading entry script of plugin ${pluginName} from ${script.src}`);
+    });
+
+    return initPromise;
+  }
+
+  /**
+   * Initialize container loaded to window scope.
+   *
+   * Required by HCC plugins.
+   * Simulates the jsonp flow used in the other type of plugins.
+   */
+  async initializeVarModule(pluginName: string) {
+    if (!this.plugins.has(pluginName)) {
+      throw new ErrorWithCause(`No plugin data exist for '${pluginName}'`);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const data = this.plugins.get(pluginName)!;
+    const moduleScope = data.manifest.name;
+    if (moduleScope) {
+      await initSharedScope(this.options.sharedScopeName);
+      const entryModule: PluginEntryModule = (
+        window as unknown as { [key: string]: PluginEntryModule }
+      )[moduleScope];
+      try {
+        await Promise.resolve(entryModule.init(this.options.sharedScope));
+      } catch (error) {
+        data.status = 'failed';
+        this.invokeListeners({
+          success: false,
+          pluginName,
+          errorCause: 'Plugin container initialization failed',
+          errorMessage: `Unable to initialize ${pluginName} webpack container! ${error}`,
+        });
+        return;
+      }
+      data.status = 'loaded';
+      data.entryCallbackFired = true;
+      this.invokeListeners({
+        success: true,
+        pluginName,
+        manifest: data.manifest,
+        entryModule,
+      });
+    } else {
+      data.status = 'failed';
+      this.invokeListeners({
+        success: false,
+        pluginName,
+        errorCause: 'Ivalid var container manifest.',
+        errorMessage: `The attribute 'name' is missing in '${pluginName}' plugin manifest!`,
+      });
+    }
+  }
+
+  setPluginData(data: PluginLoadData) {
+    this.plugins.set(data.manifest.name, data);
+  }
+
+  static isUniqueScript(
+    scriptURL: string,
+    getDocument: () => typeof document = _.constant(document),
+  ) {
+    return getDocument().head.querySelectorAll(`script[src="${scriptURL}"]`).length === 0;
```

Comment:
If we're looking for the presence of a `<script>` element with the given `src` we could simply use `querySelector` which returns the first matching element, or `null` if not found.